sessions/2018/november/Session_17.md

```diff
@@ -0,0 +1,79 @@
+
+#### Backend
+
+##### Groovy
+
+The backend application was written with a service in [Groovy][11] which is a language based in Java but with several advantages such as simpler variable definition, not need of final semicolons and several useful methods such as `bytes` as is shown following:
+
+```groovy
+final String stringRepresentation = new String(new File("src/main/resources/passwords.txt").bytes)
+```
+
+In this case that piece of code is extracting the content of the `passwords.txt` as bytes and and placing it in a `String ` variable.
+
+Groovy is a good language to learn but is dependant on be updated with Java versions, because Java is the dominant programming language.
+
+Other languaged based o Java are Kotlin and Scala.
+
+
+##### JavaScript vs CofeeScript
+
+Another example of a Programming Languaje based in another is how CofeeScript and TypeScript whic are based on javaScript ([some details here][4]). About [CofeeScript][10], a real life example was gave by Roger in how the things can complicate with CofeeScript. When CofeeScript born at the end of 2009, it was intended to be a solution to all JavaScript problems at that time, but also introduced syntax changes. With the pass of the time, JavaScript solved the problems had and improved but in a different way CofeeScript did, so this made separate paths between both of them plus all the syntax differences. Some of the arguments against CofeeScript can be found [here][5] and [here][6].
+
+##### Domain Specific Language   
+
+Domain Specific Languages (DSL) are small programming languages used inside an application with the purpose to execute certain focused tasks into applications, examples of them are SQL, CSS, Regular Expressions. Gradle scripts written with Groovy are other exampke if a DLS, that means an DLS can be written in another programming language that isn't a DSL.
+
+##### Java vs JavaScript comparison
+
+Was stated that Java produce much more enterprise ready applications because its multithreading support.
+
+Very good things can been achieved with JavaScript in the backend and the fronted -besides of the fact to be using the same programming language for both of them- but with JavaScript there is a need that the programmer deal beforehand with the problems can arise with a single threaded JavaScript application, such as blocking and long time of response when making operations. In general callback functions (that could leads to what is known as [callback hell][9]), [promises][7], [run-parallel][12] and [run-series][13] are used as alternatives to overcome the breaking application risks (whitout be a solution for single threading), but additional to that, we should too think in horizontal scaling and load balancing to have a enterprise grade application that behaves similarly as a backend Java application.
+
+So, at the end Java is simpler and safer to use to build stable backend applications.
```

Comment:
I wouldn't say Java is simpler to use. Node is certainly simpler - which is a big selling point. But Java will probably do better in the long term.